src/s3/getobject.cpp

```diff
@@ -112,14 +171,18 @@ asio::awaitable<void> irods::s3::actions::handle_getobject(
                 }
                 catch (boost::system::system_error& e) {
                     if (e.code() != beast::http::error::need_buffer) {
-                        std::cout << "Not a good error!" << std::endl;
+                        std::cout << "Not a good error!" << e.code() << std::endl;
                         throw e;
                     }
                     else {
                         // It would be nice if we could figure out something a bit more
                         // semantic than catching an exception
                     }
                 }
+
+                if (offset > range_end) {
+                    break;
+                }
```

Comment:
Consider adding a comment explaining this.

Reply:
Added comment.